test/e2e/scripts/utils.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+##### A script to generate a unique namespace #####
+##### and a parameters file for a workflow    #####
+
+
+##### Exit on error #####
+set -e
+
+##### Utility functions #####
+
+# Usage: generate_namespace <workflow_group> <workflow>
+generate_namespace()
+{
+    # Generate unique namespace from workflow_group and workflow
+    # namespace format: <workflow_group>-<workflow>-<firs_5_chars_of_prefix_check_sum>-<random_5_digits>
+    echo 'Info: Generating namespace...'
+    PREFIX=$1-$2
+    # `_` and `.` are not allowed in namespace names, replace them with `-`
+    PREFIX=${PREFIX//[_.]/-} 
+    CHECK_SUM=$(echo -n $PREFIX | md5sum | cut -c1-5)
+    SUFFIX=$RANDOM
```

Comment:
Two insights here:
- Where is the value of `RANDOM` set?
- If we choose to source that utils file in scripts that need the namespace value instead of in an "entry point" script that pass the value to other scripts, how will we be able to invoke that function without having two different results returned if there's a random part? I think either we consider that the value has been generated already and passed to `generate-parameters.sh` or `generate_namespace` should be indempotent.

Reply:
- `$RANDOM is an internal Bash function (not a constant) that returns a pseudorandom integer in the range 0 - 32767.` source: http://tldp.org/LDP/abs/html/randomvar.html

- we can make `generate_namespace` idempotent by removing the random suffix, this requires not having workflows with the same names in a workflow_group, is this guaranteed by design?

Reply:
- Nice, didn't know about this one 👍.
- It is, as we can't have two folders with the same name. However we should use a larger portion of the checksum to avoid collisions.